### ISSUE TITLE
Toolbar slows down entire app

### ISSUE THREAD

**Issue by Unforgiven-wanda** [xotahal/react-native-material-ui#290]
Hello,
I am using the Toolbar component in conjunction with `react-navigation` but I notice that doing so slows down visibly (even in production) the whole app. The animations become choppy and the app overall seems less responsive.

My code: 
```javascript
const CustomHeader = (props) => {

  const {getScreenDetails, scene, navigation} = props;
  const details = getScreenDetails(scene);
  const showBackButton = (navigation.state.routes.length > 1) ? true : false;
  const {title, searchable, ...other} = details.options;

  return (
    <Toolbar
      leftElement={showBackButton ? 'arrow-back' : 'menu'}
      onLeftElementPress={() => showBackButton ? navigation.goBack(null) : props.navigation.navigate('DrawerOpen')}
      centerElement={details.options.title}
      searchable={searchable}
      {...other}
    />
  )
};

const PrimaryNav = StackNavigator({
// ## My nav ...
}, {
// ## Other options
contentComponent: CustomDrawerContentComponent
})
```

I have noticed that this happens only with the toolbar component, as other components of this library do not have this effect.
What could be causing this? How to remedy it?

**TL;DR:** I'm dropping frames like nobody's business

Thank you.

**Comment by Angelk90** [comment]
@Unforgiven-wanda : 
It seems that it happens to me, if for example I click on a buttone that is in the toolbar, sometimes it seems that everything is blocked.
Or when I open the search, the effect seems to slow down in an imprudent way both in opening and closing.
The animation effect of the change of the arrow of the input search is also problematic.

Does it happen to you too?

**Comment by Unforgiven-wanda** [comment]
@Angelk90 Yes, that too.

**Comment by xotahal** [comment]
Does it happen in demo app? Or only in your app?

**Comment by Unforgiven-wanda** [comment]
@xotahal  Haven't been able to run the demo app (even the forked ones in the pull requests)

**Comment by Angelk90** [comment]
@xotahal :

It happens to me when I use it with the tabNavigator, it slows down the effect of opening and closing the searchable. P. S. You could answer the question: #289

**Comment by xotahal** [comment]
Hmm, there has to a lot of work in JS thread if it's slow down the effect. It is hard to say what causes that.

**Comment by Mukeyii** [comment]
I have the same issue on my Galaxy S7. I have to wait around 5 seconds when i click on a button or choose an item from the a list (since I've implemented the Toolbars).

**Comment by xotahal** [comment]
Remove all props of toolbar and let me know if the app is still slow. (After you select sth)

**Comment by Unforgiven-wanda** [comment]
@Mukeyii Care to elaborate?

**Comment by Mukeyii** [comment]
@Unforgiven-wanda I was using react-native-router-flux and whenever I switched between scenes or updated my redux store, all the other "invisible" scenes got updated and rerendered too. For this reason, the interactions and navigation became extremely slow and choppy. Now I don't allow invisible scenes to update if they're not the current one and my application is super fast and materialistic :)

**Comment by Angelk90** [comment]
@Mukeyii :  I would be interested to see how you managed the whole thing is a private project?

**Comment by Mukeyii** [comment]
It's private, but just add sth. like this to every scene/component:

```
import { Actions } from 'react-native-router-flux';

class MYCOMPONENT extends Component {

    shouldComponentUpdate() {
        if (Actions.currentScene === "MYCOMPONENT-SCENE-NAME") {
            return true;
        } 
        return false;
    }
```

**Comment by xotahal** [comment]
It is nothing to do with this library. But anyway, it's not the best fix what you can do. It is workaround how to quick fix your code. You should rather make your redux store immutable (you can use for example immutable.js - or of course, you can approach immutability also without the immutable.js) and than you should ensure that your components are re-rendered only if thay have to be re-rendered - and they should be re-rendered only if the data, which are shown on particular component, is changed ...

You can also check RN's `runAfterInteractions`
